substrate-node/pallets/pallet-runtime-upgrade/src/lib.rs

```diff
@@ -1,35 +1,35 @@
 #![cfg_attr(not(feature = "std"), no_std)]
 
-use frame_support::{
-    weights::{Pays},
-    decl_module, decl_event, traits::{EnsureOrigin},
-    dispatch::DispatchResultWithPostInfo,
-};
-
-use sp_std::prelude::*;
-
-pub trait Config: frame_system::Config {
-    type Event: From<Event> + Into<<Self as frame_system::Config>::Event>;
-    /// Origin for runtime upgrades
-    type SetCodeOrigin: EnsureOrigin<Self::Origin>;
-}
-
-decl_event!(
-	pub enum Event {}
-);
-
+pub use pallet::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+    use frame_support::pallet_prelude::*;
+    use frame_system::pallet_prelude::*;
+    use sp_std::vec::Vec;
+
+    #[pallet::pallet]
+    #[pallet::generate_store(pub(super) trait Store)]
+    #[pallet::without_storage_info]
+    pub struct Pallet<T>(_);
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
+        /// Origin for runtime upgrades
+        type SetCodeOrigin: EnsureOrigin<Self::Origin>;
+    }
 
-decl_module! {
-    pub struct Module<T: Config> for enum Call where origin: T::Origin {
-        fn deposit_event() = default;
+    #[pallet::event]
+    pub enum Event<T> {}
```

Comment:
No need to specify `T` here, since it has no generics. So you also won't need to specify this in `runtime/lib.rs`

Reply:
see errors:

![image](https://user-images.githubusercontent.com/73958772/186484305-b6f7b159-7ad9-4ad6-aa73-230a624f6977.png)

Reply:
Can we remove the pallet event if there aren't any?

Reply:
I tried but it needs to be there even if empty 
I also read it somewhere let me find it

Reply:
here:
https://docs.substrate.io/build/events-and-errors/

> The Event enum needs to be declared in your runtime's configuration trait.

Reply:
yes, but I think we don't need the generic `T`. Take a look at this example: https://github.com/paritytech/substrate/blob/1c2cdfc82be3970cf4019fd46932454d0cc4cf4a/frame/session/src/lib.rs#L544

Reply:
Hmm ok, I ll look for it

Reply:
Managed to remove Event type from pallet !!